src/extension/kernel.ts

```diff
@@ -691,6 +692,7 @@ export class Kernel implements Disposable {
     }
 
     TelemetryReporter.sendTelemetryEvent('cell.startExecute')
+    getEventReporter().reportExecution(cell)
```

Comment:
@sourishkrout I'm replying to your comment 
https://github.com/stateful/vscode-runme/pull/1602#pullrequestreview-2273121720
about adding an await here;  on the original PR.

> Yes. I incorrectly assumed the noop event reporter would be unless the "AI experiment" is turned on.

That was the intent. Is there a bug here that I need to fix? This is how I intended it to work.

The reporter should be the null op reporter by default 
https://github.com/stateful/vscode-runme/pull/1589/files#diff-f0a5c022488c3922874e791475fce22933ea551ffbc4a37c414de2831f8af593R89

and then when the extension loads if the AutoCellExperiment is turned on it get changed to the actual event reproter.
https://github.com/stateful/vscode-runme/pull/1589/files#diff-d1df40b876292a3e89a6d7d6d6e7496db98fa7ad03028b3b0f78bc90487749bdR30

One potential issue I see is that if you disable AIAutoCell the reporter won't be switched back to the null op reporter until the extension is reloaded. I assume that can be triggered by doing a reload window?
That's not great but I didn't know how to subscribe to notifications when options changed and I figured its probably good enough for now. 

> I can rewrite this reasonably quickly at a later point. Wdyt @jlewi?
If your good with the current implementation then I'm happy.

For users trying out AI and Foyle I think this fixes a big UX issue. In particular, this will remove the need to explicitly add the location of the RunMe logs to the Foyle config. This also means we can turn on learning by default in Foyle (AiAutoCell would still be disabled by default in RunMe). So I see this is as a strict improvement.

The risk is that reporting cell executions causes a degraded experience. I've tried to mitigate that with the NullOp reporter. If users don't enable AI via the AutoCell option then no additional logic (modulo the actuall noop invocation) should be invoked.

Reply:
Let's add a `// todo(sebastian): rewrite to use non-blocking impl` on top of the `reportExecution ` call and merge for now. 👍 

K, my bad. I turned off AI logs not `aiAutoCell`. In any case, as long as it's not default we're good to go.

Re, settings requiring a hard reload: that's consistent with all other settings and not something to worry about in this context. Ghost cell completing might be something we want to move into the notebook toolbar and have a setting for the default state—however, one thing at a time.

Reply:
Here we go: https://github.com/stateful/vscode-runme/pull/1602/commits/bc2b0ac881d9008eb8c671a1645a37a758b4a083

Reply:
@sourishkrout Thanks. I patched it. Your patch has an `await in it`. Did you mean to include the await?